File: src/main/java/ru/nsu/fit/g20204/kuznetsov/wireframe/scene_viewer/ToolBar.java
package ru.nsu.fit.g20204.kuznetsov.wireframe.scene_viewer;

import javax.swing.*;
import java.awt.event.ActionListener;
import java.util.HashMap;

public class ToolBar extends JToolBar {
    public ToolBar(HashMap<String, ActionListener> actions) {
        super();
        this.setFloatable(false);

        JButton openButton = new JButton("Open scene");
        this.add(openButton);
        openButton.addActionListener(actions.get("Open"));

        JButton saveButton = new JButton("Save button");
        this.add(saveButton);
        saveButton.addActionListener(actions.get("Save"));

        JButton editorButton = new JButton("BSpline editor");
        this.add(editorButton);
        editorButton.addActionListener(actions.get("BSpline editor"));

        JButton normalizeViewButton = new JButton("Normalize view");
        this.add(normalizeViewButton);
        normalizeViewButton.addActionListener(actions.get("Normalize view"));
    }
}

File: src/main/java/ru/nsu/fit/g20204/kuznetsov/wireframe/volume/scene_viewer/ToolBar.kt
package ru.nsu.fit.g20204.kuznetsov.wireframe.volume.scene_viewer

import java.awt.event.ActionListener
import javax.swing.JButton
import javax.swing.JToolBar

class ToolBar(actions: HashMap<String, ActionListener>) : JToolBar() {
    init {
        this.isFloatable = false
        val openButton = JButton("Open scene")
        this.add(openButton)
        openButton.addActionListener(actions["Open"])
        val saveButton = JButton("Save scene")
        this.add(saveButton)
        saveButton.addActionListener(actions["Save"])
        val editorButton = JButton("BSpline editor")
        this.add(editorButton)
        editorButton.addActionListener(actions["BSpline editor"])
        val normalizeViewButton = JButton("Normalize view")
        this.add(normalizeViewButton)
        normalizeViewButton.addActionListener(actions["Normalize view"])
    }
}
